<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Julien" />

<meta name="date" content="2020-04-22" />

<title>Choose your intergenic release</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BgeeCall</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="general.html">Software &amp; Data</a>
</li>
<li>
  <a href="classes_description.html">BgeeCall classes</a>
</li>
<li>
  <a href="intergenic_releases.html">Intergenic releases</a>
</li>
<li>
  <a href="one_library.html">One library</a>
</li>
<li>
  <a href="list_libraries.html">List of libraries</a>
</li>
<li>
  <a href="analysis.html">Processing data</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jwollbrett/BgeeCall_practical">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Choose your intergenic release</h1>
<h4 class="author">Julien</h4>
<h4 class="date">2020-04-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>BgeeCall_practical/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200421code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200421)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200421code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200421)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjwollbrettBgeeCallcoursetreeab08fb3c8689cb21df8c766f7686e9ac2bc5cf82targetblankab08fb3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jwollbrett/BgeeCall_course/tree/ab08fb3c8689cb21df8c766f7686e9ac2bc5cf82" target="_blank">ab08fb3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjwollbrettBgeeCallcoursetreeab08fb3c8689cb21df8c766f7686e9ac2bc5cf82targetblankab08fb3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jwollbrett/BgeeCall_course/tree/ab08fb3c8689cb21df8c766f7686e9ac2bc5cf82" target="_blank">ab08fb3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/intergenic_releases.Rmd
    Modified:   analysis/one_library.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/intergenic_releases.Rmd</code>) and HTML (<code>docs/intergenic_releases.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jwollbrett/BgeeCall_course/blob/ab08fb3c8689cb21df8c766f7686e9ac2bc5cf82/analysis/intergenic_releases.Rmd" target="_blank">ab08fb3</a>
</td>
<td>
Julien
</td>
<td>
2020-04-22
</td>
<td>
update intergenic releases page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jwollbrett/BgeeCall_course/ab08fb3c8689cb21df8c766f7686e9ac2bc5cf82/docs/intergenic_releases.html" target="_blank">ab08fb3</a>
</td>
<td>
Julien
</td>
<td>
2020-04-22
</td>
<td>
update intergenic releases page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jwollbrett/BgeeCall_course/d93bad7d16492a691076e38557227308f8350bfc/docs/intergenic_releases.html" target="_blank">d93bad7</a>
</td>
<td>
Julien
</td>
<td>
2020-04-22
</td>
<td>
update all html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jwollbrett/BgeeCall_course/blob/454071e069164efeffe6f23bcfd580e75e48aaf3/analysis/intergenic_releases.Rmd" target="_blank">454071e</a>
</td>
<td>
Julien
</td>
<td>
2020-04-22
</td>
<td>
create backbone of the website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jwollbrett/BgeeCall_course/454071e069164efeffe6f23bcfd580e75e48aaf3/docs/intergenic_releases.html" target="_blank">454071e</a>
</td>
<td>
Julien
</td>
<td>
2020-04-22
</td>
<td>
create backbone of the website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Genes called present in an RNA-Seq library are often defined using an abitrary cutoff of log2(TPM) = 1. BgeeCall allows to calculate a library specific cutoff. Reference intergenic sequences are the key element to define this library secific TPM cutoff. The cutoff is defined with the equation :</p>
<p><span class="math display">\[ \frac {proportion\ of\ reference\ intergenic\ present}{proportion\ of\ protein\ coding\ present} = 0.05 \]</span></p>
<p>This section describes the different releases of reference intergenic that are available and explain how to get information of species available in each release. <strong>In this practical we will only use the last official Bgee intergenic release.</strong></p>
</div>
<div id="list-all-releases-available" class="section level2">
<h2>List all releases available</h2>
<p>To list all available intergenic releases run the function :</p>
<pre class="r"><code>releases &lt;- list_intergenic_release()</code></pre>
<pre><code>Downloading release information of reference intergenic sequences...</code></pre>
<pre class="r"><code>subset(releases, select=c(release, releaseDate, description))</code></pre>
<pre><code>    release releaseDate
1       0.1  2018-12-21
2       0.2  2019-02-07
3 community  2019-07-22
4    custom  2019-07-22
                                                                                                                                                                                                    description
1                                                                                                                                                                  intergenic regions used to generate Bgee 14.
2                                                                          cleaned intergenic sequences based on release 0.1 (remove blocks of Ns longer than 100 and sequences containing more than 5% of Ns).
3                                                                           Release allowing to access to all reference intergenic sequences generated by the community and not present in Bgee for the moment.
4 Release allowing to use your own FASTA reference intergenic sequences. When this release is selected BgeeCall will use the sequences at UserMetadata@custom_intergenic_path to generate present/absent calls.</code></pre>
<p>In this list the first line always correspond to last official Bgee release.</p>
</div>
<div id="bgee-releases" class="section level2">
<h2>Bgee releases</h2>
<p>All releases for which the name is a number are Bgee releases. It means all reference intergenic sequences from these releases have been generated by the Bgee team. It is possible to list all species available in one bgee release with the commands :</p>
<pre class="r"><code># create an object of the class BgeeMetadata using Bgee release 0.2
bgee &lt;- new(&quot;BgeeMetadata&quot;)</code></pre>
<pre><code>
Querying Bgee to get intergenic release information...</code></pre>
<pre class="r"><code># list species available for this release
list_bgee_ref_intergenic_species(bgee)</code></pre>
<pre><code>   speciesId              speciesName numberOfLibraries   genomeVersion
1       9606             Homo sapiens              5026       GRCh38.p5
2      10090             Mus musculus               133       GRCm38.p4
3       9544           Macaca mulatta                90         MMUL1.0
4       7955              Danio rerio                67          GRCz10
5       8364       Xenopus tropicalis                66          JGI4.2
6       6239   Caenorhabditis elegans                50        WBcel235
7       9031            Gallus gallus                45         Galgal4
8      10116        Rattus norvegicus                36        Rnor_6.0
9       9913               Bos taurus                33          UMD3.1
10     13616    Monodelphis domestica                19         monDom5
11      9258 Ornithorhynchus anatinus                17           OANA5
12      7240      Drosophila simulans                17 GCA_000259055.1
13      9598          Pan troglodytes                15      CHIMP2.1.4
14      7237 Drosophila pseudoobscura                14 GCA_000001765.2
15      7227  Drosophila melanogaster                14           BDGP6
16      9593          Gorilla gorilla                13       gorGor3.1
17      9597             Pan paniscus                12      CHIMP2.1.4
18      9823               Sus scrofa                10     Sscrofa10.2
19     10141          Cavia porcellus                 9 Felis_catus_6.2
20      9685              Felis catus                 9         cavPor3
21      7230    Drosophila mojavensis                 8         EquCab2
22      9796           Equus caballus                 8 GCA_000005175.1
23      9986    Oryctolagus cuniculus                 6         eriEur1
24      9615   Canis lupus familiaris                 6       CanFam3.1
25      9365      Erinaceus europaeus                 6       OryCun2.0
26      7244       Drosophila virilis                 4 GCA_000005245.1
27     28377      Anolis carolinensis                 4       AnoCar2.0
28      7217     Drosophila ananassae                 4 GCA_000005975.1
29      7245        Drosophila yakuba                 4 GCA_000005115.1</code></pre>
</div>
<div id="community-release" class="section level2">
<h2>Community release</h2>
<p>If your species of interest is not present in the last official intergenic release you can check if the BgeeCall community already generated the reference intergenic sequences. To do so use the commands :</p>
<pre class="r"><code>list_community_ref_intergenic_species()</code></pre>
<pre><code>  speciesId numberOfLibraries annotationVersion genomeVersion kallistoVersion
1     10036                15         MesAur1.0     MesAur1.0          0.46.0
2     13686               243            Si_gnG        Si_gnG          0.44.0
                                                                                     url
1 https://zenodo.org/api/files/f46c7de0-d9a5-4ffd-a30e-4b08121ba446/ref_intergenic.fa.gz
2 https://zenodo.org/api/files/5492ff2f-91a3-4101-8d67-78b8f8625cc6/ref_intergenic.fa.gz</code></pre>
<p>It is possible to use these species in BgeeCall with the commands :</p>
<pre class="r"><code>bgee &lt;- new(&quot;BgeeMetadata&quot;, intergenic_release=&quot;community&quot;)</code></pre>
<pre><code>
Querying Bgee to get intergenic release information...</code></pre>
<pre><code>IMPORTANT : These reference intergenic sequences have not been generated by Bgee. Use with caution.</code></pre>
<p>If you did not find your species, you can participate to the community and generate your own reference intergenic sequences. A tutorial describing prerequisites and all steps is available <a href="https://github.com/BgeeDB/reference_intergenic_standalone">here</a></p>
</div>
<div id="custom-release" class="section level2">
<h2>Custom release</h2>
<p>This release allows not to download reference intergenic sequences but import them from a file.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=fr_FR.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=en_GB.UTF-8   
 [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BgeeCall_1.2.1  workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3                  lattice_0.20-41            
 [3] prettyunits_1.1.1           Rsamtools_2.2.3            
 [5] Biostrings_2.54.0           assertthat_0.2.1           
 [7] rprojroot_1.3-2             digest_0.6.25              
 [9] BiocFileCache_1.10.2        R6_2.4.1                   
[11] GenomeInfoDb_1.22.0         backports_1.1.5            
[13] stats4_3.6.3                RSQLite_2.2.0              
[15] evaluate_0.14               httr_1.4.1                 
[17] pillar_1.4.3                zlibbioc_1.32.0            
[19] rlang_0.4.5                 GenomicFeatures_1.38.2     
[21] progress_1.2.2              curl_4.3                   
[23] whisker_0.4                 blob_1.2.1                 
[25] S4Vectors_0.24.3            Matrix_1.2-18              
[27] rmarkdown_2.1               BiocParallel_1.20.1        
[29] stringr_1.4.0               RCurl_1.98-1.1             
[31] bit_1.1-15.2                biomaRt_2.42.0             
[33] DelayedArray_0.12.2         compiler_3.6.3             
[35] httpuv_1.5.2                rtracklayer_1.46.0         
[37] xfun_0.12                   pkgconfig_2.0.3            
[39] askpass_1.1                 BiocGenerics_0.32.0        
[41] htmltools_0.4.0             tximport_1.14.0            
[43] openssl_1.4.1               tidyselect_1.0.0           
[45] SummarizedExperiment_1.16.1 tibble_2.1.3               
[47] GenomeInfoDbData_1.2.2      matrixStats_0.55.0         
[49] IRanges_2.20.2              XML_3.99-0.3               
[51] crayon_1.3.4                dplyr_0.8.4                
[53] dbplyr_1.4.2                later_1.0.0                
[55] GenomicAlignments_1.22.1    bitops_1.0-6               
[57] rappdirs_0.3.1              grid_3.6.3                 
[59] jsonlite_1.6.1              DBI_1.1.0                  
[61] git2r_0.26.1                magrittr_1.5               
[63] stringi_1.4.6               XVector_0.26.0             
[65] fs_1.3.2                    promises_1.1.0             
[67] vctrs_0.2.3                 Rhdf5lib_1.8.0             
[69] tools_3.6.3                 bit64_0.9-7                
[71] Biobase_2.46.0              glue_1.3.1                 
[73] purrr_0.3.3                 hms_0.5.3                  
[75] parallel_3.6.3              yaml_2.2.1                 
[77] rhdf5_2.30.1                AnnotationDbi_1.48.0       
[79] GenomicRanges_1.38.0        memoise_1.1.0              
[81] knitr_1.28                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
