---
title: "Downstream analysis"
author: 
- name: Julien Wollbrett
  email: bgee[at]sib.swiss
  affiliation:
    &SIB Swiss Institut of Bioinformatics (SIB), Université de Lausanne 
date: "2020-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Once BgeeCall generated gene experssion calls it is possible to use them to do downstream analysis.
This section contains R code examples of potential analysis to run on present genes.

## PCA


```{r prepare data, eval=FALSE}
# read file with gene as rows and libraries as columns
file <- "PATH_TO_PRESENT_TPMS_FILE"
present_TPMs <- read.table(file = file, header = TRUE, sep = "\t")
```
```{r hidden_PCA_files, echo=FALSE}
file <- "../../temp/BgeeCall_project/input_files/downstream_analysis/present_TPMs.tsv"
present_TPMs <- read.table(file = file, header = TRUE, sep = "\t")
```
```{r run_PCA}
# transpose data.frame to have genes as columns
data_for_PCA <- t(present_TPMs)
# calculate matrix of dissimilarities
# k = the maximum dimension of the space which the data are to be represented in; must be in {1, 2, …, n-1}
# eig = indicates whether eigenvalues should be returned
mds <- cmdscale(dist(data_for_PCA), k=3, eig=TRUE)
proportion_eig <- mds$eig * 100 / sum(mds$eig)

# plot proportion of variance explained by each dimension
barplot(proportion_eig, las=1, ylim=c(0,100), xlab="Dimensions", ylab="Proportion of explained variance (%)",
        y.axis=NULL, col="darkgrey", main = "Proportion of variance explained by each dimension")

#PCA plot for the 2 first dimensions
#plotMDS(present_TPMs,xlab = "Dimension 1")
plot(mds$points[,1], -mds$points[,2], type="n", xlim=c(-2.5e+05,2.5e+05), xlab="Dimension 1", ylab="Dimension 2",
     main="PCA plot of the 2 principal dimensions", )
text(mds$points[,1], -mds$points[,2], rownames(mds$points), cex=0.8)
```

## Diffrential expression

```{r load_edgeR}
library(edgeR)
```
```{r fake_load_PCA, eval=FALSE}
# read file with gene as rows and libraries counts as columns
file_counts <- "PATH_TO_PRESENT_COUNTS_FILE"
present_counts <- read.table(file = file_counts, header = TRUE, sep = "\t")
#read file with library annotations
file_annotations <- "PATH_TO_LIBRARY_ANNOTATIONS_FILE"
library_annotations <- read.table(file = file_annotations, header = TRUE, sep = "\t")
```
```{r hidden_initialize_data, echo=FALSE}
file_counts <- "/home/julien/Documents/temp/BgeeCall_project/input_files/downstream_analysis/present_counts.tsv"
present_counts <- read.table(file = file_counts, header = TRUE, sep = "\t")
file_annotations <- "/home/julien/Documents/temp/BgeeCall_project/input_files/downstream_analysis/library_annotations.tsv"
library_annotations <- read.table(file = file_annotations, header = TRUE, sep = "\t")
```
```{r run_dif_expression}
# create list of grouping parameters (here sex)
group <- NULL
for(i in seq(colnames(present_counts))){
  group[i] <- as.character(library_annotations[library_annotations$Library.ID ==colnames(present_counts)[i],]$Sex)
}

# create DGEList object
dge <- DGEList(present_counts, group=group)
#calculate normalization factor between libraries
dge <- calcNormFactors(dge)
#estimate common and tag wise dispersion
dge <- estimateCommonDisp(dge)
dge <- estimateTagwiseDisp(dge)
#perform an exact test for the difference in expression male-female
dgeTest <- exactTest(dge)

# retrieve all gene differentially expressed with a p-value lower than 0.01
topTags <- topTags(dgeTest, n=nrow(dgeTest$table), p.value = 0.05)
#filter for FDR < 0.05
results <- topTags$table[topTags$table$FDR<0.05,]
write.table(x = results, file = "dif_expressed_genes.tsv", row.names = TRUE, quote = FALSE, sep="\t")
up_regulated_genes_female <- results[results$logFC < -1,]
up_regulated_genes_male <- results[results$logFC > 1,]

# generate an MA plot with 1% differentially expressed genes
plotSmear(dgeTest, de.tags = rownames(topTags$table)[which(topTags$table$FDR<0.01)],)

# generate a volcano plot
volcanoData <- cbind(topTags$table$logFC, -log10(topTags$table$PValue))
colnames(volcanoData) <- c("logFC", "negLogPval")
plot(volcanoData, pch=19)
```
