---
title: "Description of BgeeCall classes"
author: "Julien"
date: "2020-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

Present/absent expression calls will be generated using objects of 3 R classes specific to the BgeeCall package.
These classes contain an important number of slots (attributes) in order to tune as much as possible how present/absent expression calls are generated.
In this section we will describe these classes and their most important slots. **slots written in bold type will be used during the practical**.

## KallistoMetadata

This class allows to tune how kallisto will be run but also how presetn/absent calls will be generated
Most important slots are :

* _**download_kallisto**_ : A logical allowing to use an already installed version of kallisto or to download a version that will be used only by BgeeCall. Default value is FALSE
* _single_end_parameters_ : kallisto parameters used to run a single end mapping. Default value is "-t 1 --single -l 180 -s 30 --bias"
* _pair_end_parameters_ : kallisto parameters used to run a pair end mapping. Default value is "-t 1 --bias"
* _ignoreTxVersion_ : logical used to remove transcript version in transcript ID if TRUE. Default is FALSE. Useful if transcript version is only present in annotation or transcriptome.

## BgeeMetadata

This class allows to tune which intergenic sequences to use
Most important slot is :

* _**intergenic_release**_ : define the intergenic release that will be used. By default last intergenic release generated by the Bgee team will be used.

## UserMetadata

Both KallistoMetadata and BgeeMetadata can be used with default values. The object of class UserMetadata is the only one it is necessary to modify. It contains information specific to the analysis the user want to run.
Most important slots are :

* _**species_id**_ : The NCBI Taxon Id of the species for which the calls will be generated
* _**rnaseq_lib_path**_ : Path to the directory of the RNA-Seq library that contains fastq files
* _transcriptome_object_ : a DNAStringSet object containing the transcriptome
* _annotation_object_ : a GRanges object containing the annotation
* _**working_path**_ : path to the directory where species specific intermediate files will be generated
* _**output_dir**_ : path to the directory where files specific to the library will be generated (calls, kallisto output, etc.)

The slot _transcriptome_object_ has to be setup with one of the methods _**setTranscriptomeFromFile()**_ or _setTranscriptomeFromObject()_  
The slot _annotation_object_ has to be setup with one of the methods _**setAnnotationFromFile()**_ or _setAnnotationFromObject()_
