---
title: "Exercices"
author: 
- name: Julien Wollbrett
  email: bgee[at]sib.swiss
  affiliation:
    &SIB Swiss Institut of Bioinformatics (SIB), Universit√© de Lausanne 
date: "2020-04-24"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

4 RNA-Seq libraries of Drosophila melanogaster just arrived in your RStudio cloud project.  
All data you need are stored in the input_files directory at the root of your project.  
You and your virtual colleague have one hour to analyse these data and escape the Zoom room.


## Exercice 1 : One library

Generate present/absent expression calls for the RNA-Seq library SRX109273.  
Examples of R commands are available in the page [One library](https://jwollbrett.github.io/BgeeCall_practical/one_library.html)  
Explanations of BgeeCall classes and their most important slots are available in the page [BgeeCall classes](https://jwollbrett.github.io/BgeeCall_practical/classes_description.html)

**Important information :**  
D. melanogaster NCBI ID : 7227   
GTF gene annotations file and fasta transcriptome file are in the directory _input_files/ensembl_   
fastq files are in the directory _input_files/fastq/SRX109273/_   
Even if the generation of kallisto index takes time, you can still manage to escape the Zoom room on time.   

* What is the TPM threshold of this library ?
* What is the proportion of protein coding genes present ?
* Have a look at the plot of abundance distribution


## Exercice 2 : List of libraries
Generate present/absent expression calls for the libraries SRX109272 using a file as input
R commands and explanation of columns of the file are available in the page [List of libraries](https://jwollbrett.github.io/BgeeCall_practical/list_libraries.html)
Explanation of BgeeCall classes and their most important slots are available in the page [BgeeCall classes](https://jwollbrett.github.io/BgeeCall_practical/classes_description.html)

**Important information :**  
A template of the tsv (Tabular Separated Values) file is available at input_files/inputFile.tsv. You can edit it in RStudio with the function file.edit("PATH_TO_FILE") or in the terminal.
This file already contains information to run BgeeCall for library SRX109273 (exercice 1). You can keep these information and run BgeeCall for 2 libraries or remove them and run BgeeCall only for library SRX109272.   
fastq files of library SRX109272 are in the directory _input_files/fastq/SRX109272/_  
Do not forget to **provide the same working_path than in Exercice 1** in order to use previously generated kallisto index  

Information about the cutoff for this library are available at _YOUR_OUTPUT_DIR/gene_cutoff_info_file.tsv_

* What is the TPM threshold of this library ?
* What is the proportion of protein coding genes present ?
* Have a look at the plot of abundance distribution  


## Exercice 3 : Principle component analysis (PCA)
Fortunatly your collaborator worked well and run BgeeCall on the 2 remaining libraries. She even merged TPM values of the 4 libraries in one file where rows correspond to genes and columns correspond to libraries.  
**More importantly, she only kept in this file genes considered as present in the 4 libraries**.  
Everything is ready for downstream analysis.
She is now counting on you to do a PCA on these data

**Important information :**  
The file generated by your collaborator is available at input_files/downstream_analysis/present_TPMs.tsv
The code to run the PCA is described in the page [Processing data](https://jwollbrett.github.io/BgeeCall_practical/analysis.html)

* Check the 2 plots generated by this code and define which libraries cluster together.
* Annotations of these libraries are available at input_files/downstream_analysis/library_annotations.tsv. How do these annotations explain the PCA?


## Exercice 4 : Differential expression
Once more your collaborator was extremely productive. She merged raw counts of the 4 libraries in one file where rows correspond to genes and columns correspond to libraries.  
**More importantly, she only kept in this file genes considered as present in the 4 libraries**.  
She is now counting on you to do a differential expression analysis on these data.  

**Important information :**  
The file generated by your collaborator is available at input_files/downstream_analysis/present_counts.tsv
The code to run the Differential expression is described in the page [Processing data]()

* Which are the top 10 most differentially expressed genes?
* How many genes are considered differentially expressed if you use a threshold of p-value=0.01 and fdr=3
* Run again the differential expression analysis using the file _input_files/downstream_analysis/present_counts.tsv_. This file contains all genes (even those with absence of expression). Do you see a difference in the results?



Congratulations!!!! 
You managed to escape the Zoom room on time. Your collaborator is extremely proud of you.  


## Optional Exercice

If you are never full of data analysis you can run a geneset enrichment analysis to have more information on gene differentially expressed



