---
title: "Generate calls for a list of RNA-Seq libraries"
author: "Julien"
date: "2020-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r load-libs, message = FALSE, warning = FALSE, echo=FALSE}
library(BgeeCall)
```

## Introduction

This page describe how to generate present/absent calls for a list of RNA-Seq library.

## BgeeMetadata and KallistoMetadata classes

These 2 classes are useful to tune how BgeeCall works. In this practical we will use default values.
No need to create an object of the class BgeeMetada as only default values will be used (last official Bgee intergenic release).
An object of the KallistoMetadata class has to be created to specify to download kallisto (as it is not installed on RStudio cloud).

```{r kallisto_object}
kallisto <- new("KallistoMetadata", download_kallisto = TRUE)
```

## File describing UserMetadata objects

BgeeCall uses a tsv file to generate present/absent calls for more than one library.
In this file each line will be used to generate one UserMetadata object.
Each column of this file correspond to the value of one slot of the object.
A template of the file is present in the root of the project (inputFile.tsv). It is possible to edit it in RStudio cloud with the command file.edit("inputFile.tsv")

```{r copy_template, eval=FALSE}
file.edit("inputFile.tsv")
```

The columns of this file are :

* species_id : The NCBI ID of the species.
* reads_size : The size of the reads of your RNA-Seq library.
* rnaseq_lib_path : Path to the directory containing fastq files of the library.
* transcriptome_path : path to the transcriptome file.
* annotation_path : path to the genome annotation file. Works with GTF of GFF3 files.
* working_path : path to the working directory where results at species level will be stored (kallisto index, transcriptome with intergenic, etc.)
* output_directory : path to output directory of the RNA-Seq library results (e.g. /path/to/output/LIBRARY_ID/)

## Generate present/absent calls

Now that all objects and files have been created it is possible to run the generation of present/absent gene expression calls with one line of code


```{r run_workflow, eval=FALSE}
generate_calls_workflow(abundanceMetadata = kallisto, userFile = "inputFile.tsv")
```
