---
title: "Generate calls for a list of RNA-Seq libraries"
author: "Julien"
date: "2020-04-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r load-libs, message = FALSE, warning = FALSE, echo=FALSE}
library(BgeeCall)
```

## Introduction

This page describe how to generate present/absent calls for a list of RNA-Seq library.

## BgeeMetadata and KallistoMetadata classes

These 2 classes are useful to tune how BgeeCall works. In this practical we will use default values.
No need to create an object of the class BgeeMetada as only default values will be used (last official Bgee intergenic release).
An object of the KallistoMetadata class has to be created to specify to download kallisto (as it is not installed on RStudio cloud).

```{r kallisto_object}
kallisto <- new("KallistoMetadata", download_kallisto = TRUE)
```

## File describing UserMetadata objects

BgeeCall uses a tsv file to generate present/absent calls for more than one library.
In this file each line will be used to generate one UserMetadata object.
Each column of this file correspond to the value of one slot of the object.
A template of the file is present in the root of the project. It is possible to edit it with the command

```{r copy_template, eval=FALSE}
file.edit("inputFile.tsv")
```

The columns of this file are :

* species_id : The NCBI ID of the species.
* run_ids : The runs of the RNA-Seq library you want to use for the generation of the calls. Allows to generate expression calls for a subset of the runs of one RNA-Seq library. If not interested by this option, leave the column empty.
* reads_size : The size of the reads of your RNA-Seq library.
* rnaseq_lib_path : Path to the directory containing all fastq files generated for this library. This directory can only contains single-end runs or paired-end runs, not a mix of them.
* transcriptome_path : path to the transcriptome file.
* annotation_path : path to the genome annotation file. Works with GTF of GFF3 files.
* working_path : path to the working directory where results will be stored. Using the same working directory for different RNA-Seq libraries of the same species will allow to reuse previously generated data like the custom transcriptome index (generated from both transcriptome and reference intergenic sequences). By default the working path is defined by the `getwd()` function and correspond to the working directory of your R session. If not interested by this option, leave the column empty.
* output_directory : Both species results and RNA-Seq libraries results are by default stored at the same place using the value of the `working_path` column. However, this column allows you to define a different directory for RNA-Seq results. For instance it allows you to save calls information directly in the RNA-Seq directory. If not interested by this option, leave the column empty.

## Generate present/absent calls

Now that all objects and files have been created it is possible to run the generation of present/absent gene expression calls with one line of code


```{r run_workflow, eval=FALSE}
generate_calls_workflow(abundanceMetadata = kallisto, userFile = "PATH_TO_FILE.tsv")
```


